torch/csrc/jit/passes/onnx.cpp

```diff
@@ -129,9 +141,10 @@ void ToONNX(std::shared_ptr<tracer::TracingState>& state) {
       py_symbolic_args[input_nr++] = obj;
     }
     // Call the symbolic function
-    py::object raw_output = py::reinterpret_steal<py::object>(
-        PyObject_CallObject(pyobj.attr("symbolic").ptr(), py_symbolic_args.ptr()));
-    if (!raw_output) throw py::error_already_set();
+    py::object onnx = py::module::import("torch.onnx");
+    // Use a little trampoline function so we can give good error messages
+    // upon argument mismatch
+    py::object raw_output = onnx.attr("run_symbolic")(op->name(), pyobj.attr("symbolic"), py_symbolic_args);
```

Comment:
This looks terribly slow, and I really can't understand why do we need a Python trampoline just to append "occurred when translating X". You can easily do that in 3 lines of C, and avoid passing control around 3 files

Reply:
Invoking symbolic functions shouldn't ever be the bottleneck for export. But fine, I'll do it in C. (One other motivation for having a Python trampoline is to ease up debugging: if you want to see what symbolic functions are being called, there's now one place you can edit without having to recompile the C++)

Reply:
Yeah, uh.... I have no idea how to implement this in three lines of C (I need string concatenation! String allocation! Tuple allocation! Attribute setting! Reraising of exceptions! And handle the lifetimes of all the Python objects correctly!)

Reply:
For posterity, I spent an hour trying to make this work out, and then gave up in the end.